hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterSafemodeService.java

```diff
@@ -161,11 +161,17 @@ protected void serviceInit(Configuration conf) throws Exception {
 
   @Override
   public void periodicInvoke() {
-    long now = Time.now();
+    long now = now();
     long delta = now - startupTime;
```

Comment:
How about to invoke `monotonicNow()` to calculate the delta?

Reply:
Thanks @Hexiaoqiao help me review it
yeah, your suggestion is right, because monotonicNow() is not affected by settimeofday or similar system clock changes, if here invoke monotonicNow() to calculate the delta will avoid the exception case.

if startupTime use monotonicNow, maybe cacheLastUpdateTime and enterSafeModeTime we should also use monotonicNow() need to be consistent

what you think?

Reply:
Absolutely yes.